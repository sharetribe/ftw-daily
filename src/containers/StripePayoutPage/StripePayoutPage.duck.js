import pick from 'lodash/pick';
import {
  createStripeAccount,
  updateStripeAccount,
  fetchStripeAccount,
} from '../../ducks/stripeConnectAccount.duck';
import { fetchCurrentUser } from '../../ducks/user.duck';

// ================ Action types ================ //

export const SET_INITIAL_VALUES = 'app/StripePayoutPage/SET_INITIAL_VALUES';
export const SAVE_PAYOUT_DETAILS_REQUEST = 'app/StripePayoutPage/SAVE_PAYOUT_DETAILS_REQUEST';
export const SAVE_PAYOUT_DETAILS_SUCCESS = 'app/StripePayoutPage/SAVE_PAYOUT_DETAILS_SUCCESS';
export const SAVE_PAYOUT_DETAILS_ERROR = 'app/StripePayoutPage/SAVE_PAYOUT_DETAILS_ERROR';

// ================ Reducer ================ //

const initialState = {
  payoutDetailsSaveInProgress: false,
  payoutDetailsSaved: false,
  fromReturnURL: false,
};

export default function reducer(state = initialState, action = {}) {
  const { type, payload } = action;
  switch (type) {
    case SET_INITIAL_VALUES:
      return { ...initialState, ...payload };

    case SAVE_PAYOUT_DETAILS_REQUEST:
      return { ...state, payoutDetailsSaveInProgress: true };
    case SAVE_PAYOUT_DETAILS_ERROR:
      return { ...state, payoutDetailsSaveInProgress: false };
    case SAVE_PAYOUT_DETAILS_SUCCESS:
      return { ...state, payoutDetailsSaveInProgress: false, payoutDetailsSaved: true };

    default:
      return state;
  }
}

// ================ Action creators ================ //

export const setInitialValues = initialValues => ({
  type: SET_INITIAL_VALUES,
  payload: pick(initialValues, Object.keys(initialState)),
});

export const savePayoutDetailsRequest = () => ({
  type: SAVE_PAYOUT_DETAILS_REQUEST,
});
export const savePayoutDetailsError = () => ({
  type: SAVE_PAYOUT_DETAILS_ERROR,
});
export const savePayoutDetailsSuccess = () => ({
  type: SAVE_PAYOUT_DETAILS_SUCCESS,
});

// ================ Thunks ================ //

export const savePayoutDetails = (values, isUpdateCall) => (dispatch, getState, sdk) => {
  const upsertThunk = isUpdateCall ? updateStripeAccount : createStripeAccount;
  dispatch(savePayoutDetailsRequest());

  return dispatch(upsertThunk(values, { expand: true }))
    .then(response => {
      dispatch(savePayoutDetailsSuccess());
      return response;
    })
    .catch(() => dispatch(savePayoutDetailsError()));
};

export const loadData = () => (dispatch, getState, sdk) => {
  // Clear state so that previously loaded data is not visible
  // in case this page load fails.
  dispatch(setInitialValues());

  return dispatch(fetchCurrentUser()).then(response => {
    const currentUser = getState().user.currentUser;
    if (currentUser && currentUser.stripeAccount) {
      dispatch(fetchStripeAccount());
    }
    return response;
  });
};
