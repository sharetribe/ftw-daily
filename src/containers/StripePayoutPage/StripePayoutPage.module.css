@import '../../styles/customMediaQueries.css';
<<<<<<< HEAD
<<<<<<< HEAD:src/containers/StripePayoutPage/StripePayoutPage.module.css

=======
>>>>>>> 6cd98133265af73b146120c2fe419a6782a6e4c3:src/containers/PasswordChangePage/PasswordChangePage.module.css
=======

>>>>>>> 6cd98133265af73b146120c2fe419a6782a6e4c3
.content {
  @media (--viewportMedium) {
    margin: 32px auto 0 auto;
    width: 100%;
    max-width: 564px;
  }

  @media (--viewportLarge) {
    margin: 0;
  }
}

.desktopTopbar,
.mobileTopbar {
  box-shadow: none;
}

.title {
  hyphens: auto;
  margin-top: 8px;
  margin-bottom: 19px;

  @media (--viewportMedium) {
    margin-bottom: 47px;
  }
}
