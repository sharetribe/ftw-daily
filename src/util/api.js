// These helpers are calling FTW's own server-side routes
// so, they are not directly calling Marketplace API or Integration API.
// You can find these api endpoints from 'server/api/...' directory

import { types as sdkTypes, transit } from './sdkLoader';
import config from '../config';
import Decimal from 'decimal.js';

export const apiBaseUrl = () => {
  const port = process.env.REACT_APP_DEV_API_SERVER_PORT;
  const useDevApiServer = process.env.NODE_ENV === 'development' && !!port;

  // In development, the dev API server is running in a different port
  if (useDevApiServer) {
    return `http://localhost:${port}`;
  }

  // Otherwise, use the same domain and port as the frontend
  return `${window.location.origin}`;
};

// Application type handlers for JS SDK.
//
// NOTE: keep in sync with `typeHandlers` in `server/api-util/sdk.js`
export const typeHandlers = [
  // Use Decimal type instead of SDK's BigDecimal.
  {
    type: sdkTypes.BigDecimal,
    customType: Decimal,
    writer: v => new sdkTypes.BigDecimal(v.toString()),
    reader: v => new Decimal(v.value),
  },
];

const serialize = data => {
  return transit.write(data, { typeHandlers, verbose: config.sdk.transitVerbose });
};

const deserialize = str => {
  return transit.read(str, { typeHandlers });
};

const post = (path, body) => {
  const url = `${apiBaseUrl()}${path}`;
  const options = {
    method: 'POST',
    credentials: 'include',
    headers: {
      'Content-Type': 'application/transit+json',
    },
    body: serialize(body),
  };
  return window.fetch(url, options).then(res => {
    const contentTypeHeader = res.headers.get('Content-Type');
    const contentType = contentTypeHeader ? contentTypeHeader.split(';')[0] : null;

    if (res.status >= 400) {
      return res.json().then(data => {
        let e = new Error();
        e = Object.assign(e, data);

        throw e;
      });
    }
    if (contentType === 'application/transit+json') {
      return res.text().then(deserialize);
    } else if (contentType === 'application/json') {
      return res.json();
    }
    return res.text();
  });
};

// Fetch transaction line items from the local API endpoint.
//
// See `server/api/transaction-line-items.js` to see what data should
// be sent in the body.
export const transactionLineItems = body => {
  return post('/api/transaction-line-items', body);
};

// Initiate a privileged transaction.
//
// With privileged transitions, the transactions need to be created
// from the backend. This endpoint enables sending the booking data to
// the local backend, and passing that to the Marketplace API.
//
// See `server/api/initiate-privileged.js` to see what data should be
// sent in the body.
export const initiatePrivileged = body => {
  return post('/api/initiate-privileged', body);
};

// Transition a transaction with a privileged transition.
//
// This is similar to the `initiatePrivileged` above. It will use the
// backend for the transition. The backend endpoint will add the
// payment line items to the transition params.
//
// See `server/api/transition-privileged.js` to see what data should
// be sent in the body.
export const transitionPrivileged = body => {
  return post('/api/transition-privileged', body);
};

// Create user with identity provider (e.g. Facebook or Google)
//
// If loginWithIdp api call fails and user can't authenticate to Flex with idp
// we will show option to create a new user with idp.
// For that user needs to confirm data fetched from the idp.
// After the confirmation, this endpoint is called to create a new user with confirmed data.
//
// See `server/api/auth/createUserWithIdp.js` to see what data should
// be sent in the body.
export const createUserWithIdp = body => {
  return post('/api/auth/create-user-with-idp', body);
};
