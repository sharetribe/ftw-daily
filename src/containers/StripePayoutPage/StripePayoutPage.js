import React from 'react';
import { bool, func, oneOf, shape } from 'prop-types';
import { compose } from 'redux';
import { connect } from 'react-redux';
import routeConfiguration from '../../routeConfiguration';
import config from '../../config';
import { createResourceLocatorString } from '../../util/routes';
import { FormattedMessage, injectIntl, intlShape } from '../../util/reactIntl';
import { ensureCurrentUser } from '../../util/data';
import { propTypes } from '../../util/types';
import { isScrollingDisabled } from '../../ducks/UI.duck';
import {
  stripeAccountClearError,
  getStripeConnectAccountLink,
} from '../../ducks/stripeConnectAccount.duck';
import {
  NamedRedirect,
  LayoutSideNavigation,
  LayoutWrapperMain,
  LayoutWrapperAccountSettingsSideNav,
  LayoutWrapperTopbar,
  LayoutWrapperFooter,
  Footer,
  Page,
  StripeConnectAccountStatusBox,
  UserNav,
} from '../../components';
import { StripeConnectAccountForm } from '../../forms';
import { TopbarContainer } from '..';
import { savePayoutDetails } from './StripePayoutPage.duck';

import css from './StripePayoutPage.module.css';

const STRIPE_ONBOARDING_RETURN_URL_SUCCESS = 'success';
const STRIPE_ONBOARDING_RETURN_URL_FAILURE = 'failure';
const STRIPE_ONBOARDING_RETURN_URL_TYPES = [
  STRIPE_ONBOARDING_RETURN_URL_SUCCESS,
  STRIPE_ONBOARDING_RETURN_URL_FAILURE,
];

// Create return URL for the Stripe onboarding form
const createReturnURL = (returnURLType, rootURL, routes) => {
  const path = createResourceLocatorString(
    'StripePayoutOnboardingPage',
    routes,
    { returnURLType },
    {}
  );
  const root = rootURL.replace(/\/$/, '');
  return `${root}${path}`;
};

// Get attribute: stripeAccountData
const getStripeAccountData = stripeAccount => stripeAccount.attributes.stripeAccountData || null;

// Get last 4 digits of bank account returned in Stripe account
const getBankAccountLast4Digits = stripeAccountData =>
  stripeAccountData && stripeAccountData.external_accounts.data.length > 0
    ? stripeAccountData.external_accounts.data[0].last4
    : null;

// Check if there's requirements on selected type: 'past_due', 'currently_due' etc.
const hasRequirements = (stripeAccountData, requirementType) =>
  stripeAccountData != null &&
  stripeAccountData.requirements &&
  Array.isArray(stripeAccountData.requirements[requirementType]) &&
  stripeAccountData.requirements[requirementType].length > 0;

// Redirect user to Stripe's hosted Connect account onboarding form
const handleGetStripeConnectAccountLinkFn = (getLinkFn, commonParams) => type => () => {
  getLinkFn({ type, ...commonParams })
    .then(url => {
      window.location.href = url;
    })
    .catch(err => console.error(err));
};

export const StripePayoutPageComponent = props => {
  const {
    currentUser,
    scrollingDisabled,
    getAccountLinkInProgress,
    getAccountLinkError,
    createStripeAccountError,
    updateStripeAccountError,
    fetchStripeAccountError,
    stripeAccountFetched,
    stripeAccount,
    onPayoutDetailsFormChange,
    onPayoutDetailsFormSubmit,
    onGetStripeConnectAccountLink,
    payoutDetailsSaveInProgress,
    payoutDetailsSaved,
    params,
    intl,
  } = props;

  const { returnURLType } = params;
  const ensuredCurrentUser = ensureCurrentUser(currentUser);
  const currentUserLoaded = !!ensuredCurrentUser.id;
  const stripeConnected = currentUserLoaded && !!stripeAccount && !!stripeAccount.id;

  const title = intl.formatMessage({ id: 'StripePayoutPage.title' });

  const formDisabled = getAccountLinkInProgress;

  const rootURL = config.canonicalRootURL;
  const routes = routeConfiguration();
  const successURL = createReturnURL(STRIPE_ONBOARDING_RETURN_URL_SUCCESS, rootURL, routes);
  const failureURL = createReturnURL(STRIPE_ONBOARDING_RETURN_URL_FAILURE, rootURL, routes);

  const accountId = stripeConnected ? stripeAccount.id : null;
  const stripeAccountData = stripeConnected ? getStripeAccountData(stripeAccount) : null;
  const requirementsMissing =
    stripeAccount &&
    (hasRequirements(stripeAccountData, 'past_due') ||
      hasRequirements(stripeAccountData, 'currently_due'));

  const savedCountry = stripeAccountData ? stripeAccountData.country : null;

  const handleGetStripeConnectAccountLink = handleGetStripeConnectAccountLinkFn(
    onGetStripeConnectAccountLink,
    {
      accountId,
      successURL,
      failureURL,
    }
  );

  const returnedNormallyFromStripe = returnURLType === STRIPE_ONBOARDING_RETURN_URL_SUCCESS;
  const returnedAbnormallyFromStripe = returnURLType === STRIPE_ONBOARDING_RETURN_URL_FAILURE;
  const showVerificationNeeded = stripeConnected && requirementsMissing;

  // Redirect from success URL to basic path for StripePayoutPage
  if (returnedNormallyFromStripe && stripeConnected && !requirementsMissing) {
    return <NamedRedirect name="StripePayoutPage" />;
  }

  // Failure url should redirect back to Stripe since it's most likely due to page reload
  // Account link creation will fail if the account is the reason
  if (returnedAbnormallyFromStripe && !getAccountLinkError) {
    handleGetStripeConnectAccountLink('custom_account_verification')();
  }

  return (
    <Page title={title} scrollingDisabled={scrollingDisabled}>
      <LayoutSideNavigation>
        <LayoutWrapperTopbar>
          <TopbarContainer
            currentPage="StripePayoutPage"
            desktopClassName={css.desktopTopbar}
            mobileClassName={css.mobileTopbar}
          />
          <UserNav selectedPageName="StripePayoutPage" />
        </LayoutWrapperTopbar>
        <LayoutWrapperAccountSettingsSideNav currentTab="StripePayoutPage" />
        <LayoutWrapperMain>
          <div className={css.content}>
            <h1 className={css.title}>
              <FormattedMessage id="StripePayoutPage.heading" />
            </h1>
            {!currentUserLoaded ? (
              <FormattedMessage id="StripePayoutPage.loadingData" />
            ) : returnedAbnormallyFromStripe && !getAccountLinkError ? (
              <FormattedMessage id="StripePayoutPage.redirectingToStripe" />
            ) : (
              <StripeConnectAccountForm
                disabled={formDisabled}
                inProgress={payoutDetailsSaveInProgress}
                ready={payoutDetailsSaved}
                currentUser={ensuredCurrentUser}
                stripeBankAccountLastDigits={getBankAccountLast4Digits(stripeAccountData)}
                savedCountry={savedCountry}
                submitButtonText={intl.formatMessage({
                  id: 'StripePayoutPage.submitButtonText',
                })}
                stripeAccountError={
                  createStripeAccountError || updateStripeAccountError || fetchStripeAccountError
                }
                stripeAccountLinkError={getAccountLinkError}
                stripeAccountFetched={stripeAccountFetched}
                onChange={onPayoutDetailsFormChange}
                onSubmit={onPayoutDetailsFormSubmit}
                onGetStripeConnectAccountLink={handleGetStripeConnectAccountLink}
                stripeConnected={stripeConnected}
              >
                {stripeConnected && !returnedAbnormallyFromStripe && showVerificationNeeded ? (
                  <StripeConnectAccountStatusBox
                    type="verificationNeeded"
                    inProgress={getAccountLinkInProgress}
                    onGetStripeConnectAccountLink={handleGetStripeConnectAccountLink(
                      'custom_account_verification'
                    )}
                  />
                ) : stripeConnected && savedCountry && !returnedAbnormallyFromStripe ? (
                  <StripeConnectAccountStatusBox
                    type="verificationSuccess"
                    inProgress={getAccountLinkInProgress}
                    disabled={payoutDetailsSaveInProgress}
                    onGetStripeConnectAccountLink={handleGetStripeConnectAccountLink(
                      'custom_account_update'
                    )}
                  />
                ) : null}
              </StripeConnectAccountForm>
            )}
          </div>
        </LayoutWrapperMain>
        <LayoutWrapperFooter>
          <Footer />
        </LayoutWrapperFooter>
      </LayoutSideNavigation>
    </Page>
  );
};

StripePayoutPageComponent.defaultProps = {
  currentUser: null,
  createStripeAccountError: null,
  updateStripeAccountError: null,
  fetchStripeAccountError: null,
  getAccountLinkError: null,
  stripeAccount: null,
  params: {
    returnURLType: null,
  },
};

StripePayoutPageComponent.propTypes = {
  currentUser: propTypes.currentUser,
  scrollingDisabled: bool.isRequired,
  getAccountLinkInProgress: bool.isRequired,
  payoutDetailsSaveInProgress: bool.isRequired,
  createStripeAccountError: propTypes.error,
  getAccountLinkError: propTypes.error,
  updateStripeAccountError: propTypes.error,
  fetchStripeAccountError: propTypes.error,
  stripeAccount: propTypes.stripeAccount,
  stripeAccountFetched: bool.isRequired,
  payoutDetailsSaved: bool.isRequired,

  onPayoutDetailsFormChange: func.isRequired,
  onPayoutDetailsFormSubmit: func.isRequired,
  onGetStripeConnectAccountLink: func.isRequired,
  params: shape({
    returnURLType: oneOf(STRIPE_ONBOARDING_RETURN_URL_TYPES),
  }),

  // from injectIntl
  intl: intlShape.isRequired,
};

const mapStateToProps = state => {
  const {
    getAccountLinkInProgress,
    getAccountLinkError,
    createStripeAccountError,
    updateStripeAccountError,
    fetchStripeAccountError,
    stripeAccount,
    stripeAccountFetched,
  } = state.stripeConnectAccount;
  const { currentUser } = state.user;
  const { payoutDetailsSaveInProgress, payoutDetailsSaved } = state.StripePayoutPage;
  return {
    currentUser,
    getAccountLinkInProgress,
    getAccountLinkError,
    createStripeAccountError,
    updateStripeAccountError,
    fetchStripeAccountError,
    stripeAccount,
    stripeAccountFetched,
    payoutDetailsSaveInProgress,
    payoutDetailsSaved,
    scrollingDisabled: isScrollingDisabled(state),
  };
};

const mapDispatchToProps = dispatch => ({
  onPayoutDetailsFormChange: () => dispatch(stripeAccountClearError()),
  onPayoutDetailsFormSubmit: (values, isUpdateCall) =>
    dispatch(savePayoutDetails(values, isUpdateCall)),
  onGetStripeConnectAccountLink: params => dispatch(getStripeConnectAccountLink(params)),
});

const StripePayoutPage = compose(
  connect(
    mapStateToProps,
    mapDispatchToProps
  ),
  injectIntl
)(StripePayoutPageComponent);

export default StripePayoutPage;
