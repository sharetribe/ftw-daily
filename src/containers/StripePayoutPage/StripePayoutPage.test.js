import React from 'react';
import { renderShallow } from '../../util/test-helpers';
import { fakeIntl, createCurrentUser, createStripeAccount } from '../../util/test-data';
import { StripePayoutPageComponent } from './StripePayoutPage';

const noop = () => null;

describe('StripePayoutPage', () => {
  it('matches snapshot with Stripe not connected', () => {
    const currentUser = createCurrentUser('stripe-not-connected');
    expect(currentUser.stripeAccount).toBeUndefined();
    const tree = renderShallow(
      <StripePayoutPageComponent
        currentUser={currentUser}
        scrollingDisabled={false}
        payoutDetailsSaveInProgress={false}
        payoutDetailsSaved={false}
        onPayoutDetailsFormChange={noop}
        onPayoutDetailsFormSubmit={noop}
        onGetStripeConnectAccountLink={noop}
        stripeAccountFetched={false}
        getAccountLinkInProgress={false}
        intl={fakeIntl}
        history={{ replace: noop }}
      />
    );
    expect(tree).toMatchSnapshot();
  });
  it('matches snapshot with Stripe connected', () => {
    const currentUser = createCurrentUser(
      'stripe-connected',
      {},
      {
        stripeAccount: createStripeAccount(),
      }
    );
    expect(currentUser.stripeAccount).toBeDefined();
    const tree = renderShallow(
      <StripePayoutPageComponent
        currentUser={currentUser}
        scrollingDisabled={false}
        payoutDetailsSaveInProgress={false}
        payoutDetailsSaved={false}
        onPayoutDetailsFormChange={noop}
        onPayoutDetailsFormSubmit={noop}
        onGetStripeConnectAccountLink={noop}
        stripeAccountFetched={false}
        getAccountLinkInProgress={false}
        intl={fakeIntl}
        history={{ replace: noop }}
      />
    );
    expect(tree).toMatchSnapshot();
  });
  it('matches snapshot with details submitted', () => {
    const currentUser = createCurrentUser(
      'stripe-connected',
      {},
      {
        stripeAccount: createStripeAccount(),
      }
    );
    expect(currentUser.stripeAccount).toBeDefined();
    const tree = renderShallow(
      <StripePayoutPageComponent
        currentUser={currentUser}
        scrollingDisabled={false}
        payoutDetailsSaveInProgress={false}
        payoutDetailsSaved={true}
        onPayoutDetailsFormChange={noop}
        onPayoutDetailsFormSubmit={noop}
        onGetStripeConnectAccountLink={noop}
        stripeAccountFetched={false}
        getAccountLinkInProgress={false}
        intl={fakeIntl}
        history={{ replace: noop }}
      />
    );
    expect(tree).toMatchSnapshot();
  });
});
