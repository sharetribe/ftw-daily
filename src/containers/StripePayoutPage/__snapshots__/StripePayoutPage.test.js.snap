// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StripePayoutPage matches snapshot with Stripe connected 1`] = `
<Page
  scrollingDisabled={false}
  title="StripePayoutPage.title"
>
  <LayoutSideNavigation
    className={null}
    containerClassName={null}
    rootClassName={null}
  >
    <LayoutWrapperTopbar
      className={null}
      rootClassName={null}
    >
      <withRouter(Connect(TopbarContainerComponent))
        currentPage="StripePayoutPage"
        desktopClassName="desktopTopbar"
        mobileClassName="mobileTopbar"
      />
      <UserNav
        className={null}
        rootClassName={null}
        selectedPageName="StripePayoutPage"
      />
    </LayoutWrapperTopbar>
    <withViewport(LayoutWrapperAccountSettingsSideNavComponent)
      currentTab="StripePayoutPage"
    />
    <LayoutWrapperMain
      className={null}
      rootClassName={null}
    >
      <div
        className="content"
      >
        <h1
          className="title"
        >
          <MemoizedFormattedMessage
            id="StripePayoutPage.heading"
          />
        </h1>
        <injectIntl(StripeConnectAccountFormComponent)
          currentUser={
            Object {
              "attributes": Object {
                "banned": false,
                "deleted": false,
                "email": "stripe-connected@example.com",
                "emailVerified": true,
                "profile": Object {
                  "abbreviatedName": "stripe-connected abbreviated name",
                  "displayName": "stripe-connected display name",
                  "firstName": "stripe-connected first name",
                  "lastName": "stripe-connected last name",
                },
              },
              "id": UUID {
                "_sdkType": "UUID",
                "uuid": "stripe-connected",
              },
              "profileImage": Object {},
              "stripeAccount": Object {
                "attributes": Object {
                  "stripeAccountId": "acc_testiaccountid",
                },
                "id": UUID {
                  "_sdkType": "UUID",
                  "uuid": undefined,
                },
                "type": "stripeAccount",
              },
              "type": "currentUser",
            }
          }
          disabled={false}
          inProgress={false}
          onChange={[Function]}
          onGetStripeConnectAccountLink={[Function]}
          onSubmit={[Function]}
          ready={false}
          savedCountry={null}
          stripeAccountError={null}
          stripeAccountFetched={false}
          stripeAccountLinkError={null}
          stripeBankAccountLastDigits={null}
          stripeConnected={false}
          submitButtonText="StripePayoutPage.submitButtonText"
        />
      </div>
    </LayoutWrapperMain>
    <LayoutWrapperFooter
      className={null}
      rootClassName={null}
    >
      <injectIntl(Footer) />
    </LayoutWrapperFooter>
  </LayoutSideNavigation>
</Page>
`;

exports[`StripePayoutPage matches snapshot with Stripe not connected 1`] = `
<Page
  scrollingDisabled={false}
  title="StripePayoutPage.title"
>
  <LayoutSideNavigation
    className={null}
    containerClassName={null}
    rootClassName={null}
  >
    <LayoutWrapperTopbar
      className={null}
      rootClassName={null}
    >
      <withRouter(Connect(TopbarContainerComponent))
        currentPage="StripePayoutPage"
        desktopClassName="desktopTopbar"
        mobileClassName="mobileTopbar"
      />
      <UserNav
        className={null}
        rootClassName={null}
        selectedPageName="StripePayoutPage"
      />
    </LayoutWrapperTopbar>
    <withViewport(LayoutWrapperAccountSettingsSideNavComponent)
      currentTab="StripePayoutPage"
    />
    <LayoutWrapperMain
      className={null}
      rootClassName={null}
    >
      <div
        className="content"
      >
        <h1
          className="title"
        >
          <MemoizedFormattedMessage
            id="StripePayoutPage.heading"
          />
        </h1>
        <injectIntl(StripeConnectAccountFormComponent)
          currentUser={
            Object {
              "attributes": Object {
                "banned": false,
                "deleted": false,
                "email": "stripe-not-connected@example.com",
                "emailVerified": true,
                "profile": Object {
                  "abbreviatedName": "stripe-not-connected abbreviated name",
                  "displayName": "stripe-not-connected display name",
                  "firstName": "stripe-not-connected first name",
                  "lastName": "stripe-not-connected last name",
                },
              },
              "id": UUID {
                "_sdkType": "UUID",
                "uuid": "stripe-not-connected",
              },
              "profileImage": Object {},
              "type": "currentUser",
            }
          }
          disabled={false}
          inProgress={false}
          onChange={[Function]}
          onGetStripeConnectAccountLink={[Function]}
          onSubmit={[Function]}
          ready={false}
          savedCountry={null}
          stripeAccountError={null}
          stripeAccountFetched={false}
          stripeAccountLinkError={null}
          stripeBankAccountLastDigits={null}
          stripeConnected={false}
          submitButtonText="StripePayoutPage.submitButtonText"
        />
      </div>
    </LayoutWrapperMain>
    <LayoutWrapperFooter
      className={null}
      rootClassName={null}
    >
      <injectIntl(Footer) />
    </LayoutWrapperFooter>
  </LayoutSideNavigation>
</Page>
`;

exports[`StripePayoutPage matches snapshot with details submitted 1`] = `
<Page
  scrollingDisabled={false}
  title="StripePayoutPage.title"
>
  <LayoutSideNavigation
    className={null}
    containerClassName={null}
    rootClassName={null}
  >
    <LayoutWrapperTopbar
      className={null}
      rootClassName={null}
    >
      <withRouter(Connect(TopbarContainerComponent))
        currentPage="StripePayoutPage"
        desktopClassName="desktopTopbar"
        mobileClassName="mobileTopbar"
      />
      <UserNav
        className={null}
        rootClassName={null}
        selectedPageName="StripePayoutPage"
      />
    </LayoutWrapperTopbar>
    <withViewport(LayoutWrapperAccountSettingsSideNavComponent)
      currentTab="StripePayoutPage"
    />
    <LayoutWrapperMain
      className={null}
      rootClassName={null}
    >
      <div
        className="content"
      >
        <h1
          className="title"
        >
          <MemoizedFormattedMessage
            id="StripePayoutPage.heading"
          />
        </h1>
        <injectIntl(StripeConnectAccountFormComponent)
          currentUser={
            Object {
              "attributes": Object {
                "banned": false,
                "deleted": false,
                "email": "stripe-connected@example.com",
                "emailVerified": true,
                "profile": Object {
                  "abbreviatedName": "stripe-connected abbreviated name",
                  "displayName": "stripe-connected display name",
                  "firstName": "stripe-connected first name",
                  "lastName": "stripe-connected last name",
                },
              },
              "id": UUID {
                "_sdkType": "UUID",
                "uuid": "stripe-connected",
              },
              "profileImage": Object {},
              "stripeAccount": Object {
                "attributes": Object {
                  "stripeAccountId": "acc_testiaccountid",
                },
                "id": UUID {
                  "_sdkType": "UUID",
                  "uuid": undefined,
                },
                "type": "stripeAccount",
              },
              "type": "currentUser",
            }
          }
          disabled={false}
          inProgress={false}
          onChange={[Function]}
          onGetStripeConnectAccountLink={[Function]}
          onSubmit={[Function]}
          ready={true}
          savedCountry={null}
          stripeAccountError={null}
          stripeAccountFetched={false}
          stripeAccountLinkError={null}
          stripeBankAccountLastDigits={null}
          stripeConnected={false}
          submitButtonText="StripePayoutPage.submitButtonText"
        />
      </div>
    </LayoutWrapperMain>
    <LayoutWrapperFooter
      className={null}
      rootClassName={null}
    >
      <injectIntl(Footer) />
    </LayoutWrapperFooter>
  </LayoutSideNavigation>
</Page>
`;
